src/core/ledger/policies/immediate.test.js

```diff
@@ -0,0 +1,132 @@
+// @flow
+
+import * as G from "../grain";
+import {processIdentities} from "../processedIdentities";
+import {random as randomUuid} from "../../../util/uuid";
+import {immediateReceipts} from "./immediate";
+
+describe("core/ledger/policies/immediate", () => {
+  describe("immediateReceipts", () => {
+    const id1 = randomUuid();
+    const id2 = randomUuid();
+    const identities = [
+      {
+        cred: [10, 20, 30, 40],
+        id: id1,
+        paid: G.ONE,
+      },
+      {
+        cred: [200, 0, 0, 0],
+        id: id2,
+        paid: G.ZERO,
+      },
+    ];
+    const processedIdentities = processIdentities(identities);
+
+    it("errors on invalid range", () => {
+      const policy = {
+        policyType: "IMMEDIATE",
+        budget: G.ONE,
+        numPeriodsLookback: -1,
+      };
+      expect(() => immediateReceipts(policy, processedIdentities)).toThrowError(
+        `numPeriodsLookback must be at least 1`
+      );
+    });
+
+    it("errors on float instead of int", () => {
+      const policy = {
+        policyType: "IMMEDIATE",
+        budget: G.ONE,
+        numPeriodsLookback: 1.5,
+      };
+      expect(() => immediateReceipts(policy, processedIdentities)).toThrowError(
+        `numPeriodsLookback must be an integer`
+      );
+    });
+
+    it("defaults lookback period > history to max history", () => {
+      const policy1 = {
+        policyType: "IMMEDIATE",
+        budget: G.ONE,
+        numPeriodsLookback: 4,
+      };
+      const policy2 = {
+        policyType: "IMMEDIATE",
+        budget: G.ONE,
+        numPeriodsLookback: 50,
+      };
+      const expected1 = immediateReceipts(policy1, processedIdentities);
+      const expected2 = immediateReceipts(policy2, processedIdentities);
+      expect(expected1).toEqual(expected2);
+    });
+
+    it("defaults missing lookback period to 1", () => {
+      const policy1 = {
+        policyType: "IMMEDIATE",
+        budget: G.ONE,
+      };
+      const policy2 = {
+        policyType: "IMMEDIATE",
+        budget: G.ONE,
+        numPeriodsLookback: 1,
+      };
+      const expected1 = immediateReceipts(policy1, processedIdentities);
+      const expected2 = immediateReceipts(policy2, processedIdentities);
+      expect(expected1).toEqual(expected2);
+    });
+
+    it("returns a valid $ReadOnlyArray of GrainReceipt's at max interval length", () => {
+      const expectedAmounts = G.splitBudget(G.ONE, [100, 200]);
+      const expected = [
+        {
+          id: id1,
+          amount: expectedAmounts[0],
+        },
+        {
+          id: id2,
+          amount: expectedAmounts[1],
+        },
+      ];
+      const policy = {
+        policyType: "IMMEDIATE",
+        budget: G.ONE,
+        numPeriodsLookback: 4,
+      };
+      expect(immediateReceipts(policy, processedIdentities)).toEqual(expected);
+    });
+
+    it("returns a valid $ReadOnlyArray of GrainReceipt's at min interval length", () => {
```

Comment:
Technically speaking, $ReadOnlyArray is a type-level designation, not a runtime designation. So at runtime, there is no read-only array, just an actual array. 

Also, does "min interval length" mean "1 period lookback"? What does "valid" mean in this case, exactly? Do we just mean it "correctly computes receipts when numPeriodsLookback is 1"?

(same questions about the test case above)

Reply:
Done.